src/Ztm.Zcoin.Rpc/ZcoinRpcClient.cs

```
@@ -362,70 +388,83 @@ static ushort ToNative(PropertyType type)
         {
-            var args = new List<object>()
-            {
-                transaction.ToHex(),
-            };
-
             // Invoke RPC.
-            var resp = await this.client.SendCommandAsync("exodus_decodetransaction", args.ToArray());
+            var resp = await this.client.SendCommandAsync("exodus_gettransaction", transaction.GetHash());
             var rawRefAddress = resp.Result.Value<string>("referenceaddress");
 
             return new TransactionInfomation{
                 TxId = uint256.Parse(resp.Result.Value<string>("txid")),
-                Fee = Money.Parse(resp.Result.Value<string>("fee")),
                 SendingAddress = BitcoinAddress.Create(resp.Result.Value<string>("sendingaddress"), this.client.Network),
                 ReferenceAddress = (rawRefAddress == null || rawRefAddress == string.Empty)
                     ? null : BitcoinAddress.Create(resp.Result.Value<string>("referenceaddress"), this.client.Network),
                 IsMine = resp.Result.Value<bool>("ismine"),
+                Confirmations = resp.Result.Value<int>("confirmations"),
+                Fee = Money.Parse(resp.Result.Value<string>("fee")),
+                BlockTime = resp.Result.Value<int>("blocktime"),
+                Valid = resp.Result.Value<bool>("valid"),
+                InvalidReason = resp.Result.Value<string>("invalidreason"),
                 Version = resp.Result.Value<int>("version"),
                 TypeInt = resp.Result.Value<int>("type_int"),
                 Type = resp.Result.Value<string>("type")
             };
         }
 
-        async Task<(string Payload, int PayloadSize)> GetPayloadAsync(Transaction transaction)
+        async Task<(string payload, int payloadSize)> GetPayloadAsync(Transaction transaction)
         {
             var resp = await this.client.SendCommandAsync("exodus_getpayload", new object[]{transaction.GetHash()});
```

We should return `null` instead if transaction is not Exodus transaction.

---

just let it throw for now, already create an issue.